File: java-pi-pantry/src/main/java/com/langrsoft/util/JsonUtil.java
package com.langrsoft.util;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

public class JsonUtil {
    public static <T> String toJson(T object) {
        try {
            return new ObjectMapper().writeValueAsString(object);
        } catch (JsonProcessingException e) {
            throw new JsonParseException(e);
        }
    }

    public static <T> T parse(String json, Class<T> type) {
        try {
            return new ObjectMapper().readValue(json, type);
        } catch (IOException e) {
            throw new JsonParseException(e);
        }
    }
}

File: csharp-pi-pantry/csharp-pi-pantry/Util/JsonUtil.cs
using Newtonsoft.Json;
using Pipantry.Domain;

namespace Pipantry.Util
{
    public class JsonUtil
    {
        public static string ToJson<T>(T obj)
        {
            try
            {
                return JsonConvert.SerializeObject(obj);
            }
            catch (JsonException e)
            {
                throw new JsonParseException(e);
            }
        }

        public static T Parse<T>(string json)
        {
            try
            {
                return JsonConvert.DeserializeObject<T>(json);
            }
            catch (JsonException e)
            {
                throw new JsonParseException(e);
            }

        }
    }
}
